docs/source/security/authentication.mdx

```diff
@@ -5,53 +5,59 @@ description: Control access to your GraphQL API
 
-* **Authentication** is determining whether a given user is logged in, and subsequently determining _which_ user someone is.
-* **Authorization** is then determining what a given user has permission to do or see.
+- **Authentication** is determining whether a given user is logged in, and subsequently determining _which_ user someone is.
+- **Authorization** is then determining what a given user has permission to do or see.
 
 ## Putting authenticated user info on the `context`
 
 Before we can correctly control access to data, we have to authenticate a user. There are many patterns for providing authentication credentials, including HTTP headers and JSON web tokens.
 
 The example below extracts a user token from the HTTP `Authorization` header included in each operation request. It then fetches the corresponding user object for that token and adds that object to [the `context` object](../data/resolvers/#the-context-argument) that's passed to every executed resolver. Each resolver can then use this object to determine what data the user has access to.
 
-```js
-const { ApolloServer } = require('apollo-server');
-const {
-  ApolloServerPluginLandingPageLocalDefault,
-} = require('apollo-server-core');
-
-const server = new ApolloServer({
- typeDefs,
- resolvers,
- csrfPrevention: true,
- cache: 'bounded',
- context: ({ req }) => {
-   // Note: This example uses the `req` argument to access headers,
-   // but the arguments received by `context` vary by integration.
-   // This means they vary for Express, Koa, Lambda, etc.
-   //
-   // To find out the correct arguments for a specific integration,
-   // see https://www.apollographql.com/docs/apollo-server/api/apollo-server/#middleware-specific-context-fields
-
-   // Get the user token from the headers.
-   const token = req.headers.authorization || '';
-
-   // Try to retrieve a user with the token
-   const user = getUser(token);
-
-   // Add the user to the context
-   return { user };
- },
- plugins: [
-    ApolloServerPluginLandingPageLocalDefault({ embed: true }),
-  ],
+<!-- TODO(AS4) Add note back in about where to find integration specific context arguments -->
+
+<MultiCodeBlock>
+
+```ts
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+
+interface MyContext {
+  // we'd define the properties a user should have
+  // in a separate user interface (e.g., email, id, url, etc.)
+  user: UserInterface;
+}
+
+const server = new ApolloServer<MyContext>({
+  typeDefs,
+  resolvers,
 });
 
-server.listen().then(({ url }) => {
- console.log(`🚀 Server ready at ${url}`)
+const { url } = await startStandaloneServer(server, {
+  listen: { port: 4000 },
+  // highlight-start
+  context: async ({ req, res }) => {
+    // Note: This example uses the `req` argument to access headers,
+    // but the arguments received by `context` vary by integration.
+    // This means they vary for Express, Koa, Lambda, etc.
```

Comment:
Let's substitute Fastify for Koa here, it's more likely we'll actually have it.

Maybe state that for startStandaloneServer it's an [http.IncomingMessage](https://nodejs.org/api/http.html#class-httpincomingmessage) ?

Reply:
Great idea, I'll do both!